### PR TITLE
Add new s3 sink documentation for Data Prepper 2.8

### LINKED ISSUE #6925: [DOC] Data Prepper S3 sink new features
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
There have been some changes to the S3 sink

1. Support for dynamic `path_prefix` to group Events to different "folders" in the bucket - (https://github.com/opensearch-project/data-prepper/pull/4346)
2. Existing `threshold` is at the group level. Support for `aggregate_threshold` to limit total memory usage for high group cardinality cases - (https://github.com/opensearch-project/data-prepper/pull/4385)
3. Support for dynamic `bucket` name to send to different buckets dynamically, including a `default_bucket` parameter that will be sent to when a `NoSuchBucketException` or access denied exception is thrown.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by graytaylor0**
### Description
Adds new documentation for new parameters in Data Prepper's s3 sink for Data Prepper 2.8 release

### Issues Resolved
Closes #6925 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Thanks, @graytaylor0 ! Is this ready for doc review? In the meantime, can you please address the style errors?

**Review by vagimeli — COMMENTED** [review]
Doc review completed. Please see feedback. Once feedback is addressed, I'll move this forward to editorial. Thank you.

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:74]:
```suggestion
When Data Prepper fetches data from an S3 bucket, it verifies bucket ownership using [bucket owner condition](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucket-owner-condition.html). 

By default, the S3 sink does not require `bucket_owners`. If `bucket_owners` is configured and a bucket is not included in one of these mapped configurations, `default_bucket_owner` defaults to the account id from `aws.sts_role_arn`. 
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:94]:
```suggestion
`bucket_owners` and `default_bucket_owner` can be used together.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:138]:
```suggestion
`flush_capacity_ratio` | No                                 | Float  | The percentage of groups to be force-flushed when `aggregate_threshold maximum_size` is reached. Default is `0.5`.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:153]:
```suggestion
`path_prefix` | No | String | The S3 key prefix path to use for objects written to S3. Accepts date-time formatting and dynamic injection of values using [Data Prepper expressions](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/). For example, you can use `/${/my_partition_key}/%{yyyy}/%{MM}/%{dd}/%{HH}/` to create hourly folders in S3 based on the `my_partition_key` value. The prefix path should end with `/`. By default, Data Prepper writes objects to the S3 bucket root.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:163]:
```suggestion
Because Avro requires a schema, you may either define the schema yourself or have Data Prepper automatically generate a schema. It is recommended that you define your schema to accurately reflect your needs.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:165]:
```suggestion
It is recommended making Avro fields use a null [union](https://avro.apache.org/docs/current/specification/#unions) to allow missing values; otherwise, ensure all required fields are present for each event. Use non-nullable fields only when certain of their presence.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:169]:
```suggestion
In cases where your data is uniform, you may be able to automatically generate a schema. Automatically generated schemas are based on the first event that the codec receives.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:191]:
```suggestion
The `parquet` codec writes events into a Parquet file. When using the codec, set `buffer_type` to `in_memory`.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:193]:
```suggestion
The `parquet` codec writes data using the schema. Because Parquet requires an Avro schema, you may either define the schema yourself or have Data Prepper automatically generate it. It is recommended that you define your schema to accurately reflect your needs.
```

**Comment by vagimeli** [comment]
@dlvenable Please provide a technical review and approval of this documentation. I'll move it into editorial once you've approved. Thank you.

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (dlvenable) [inline _data-prepper/pipelines/configuration/sinks/s3.md:203]: Can we say "We recommended" or "The Data Prepper maintainers recommend" instead of "It is recommended." The former seems non-committal as if we don't want to stand behind our recommendation.

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:170]:
```suggestion
The `avro` codec writes an event as an [Apache Avro](https://avro.apache.org/) document. Because Avro requires a schema, you may either define the schema or have Data Prepper automatically generate it. Defining your own schema is recommended, as this will allow it to be tailored to your particular use case.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:176]:
```suggestion
Avro fields should use a null [union](https://avro.apache.org/docs/current/specification/#unions), as this will allow missing values. Otherwise, all required fields must be present for each event. Use non-nullable fields only when you are certain they exist.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:143]:
```suggestion
`flush_capacity_ratio` | No                                 | Float  | The percentage of groups to be force-flushed when `aggregate_threshold maximum_size` is reached. Percentage is expressed from `0.0`--`1.0`. Default is `0.5`.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:80]:
```suggestion
By default, the S3 sink does not require `bucket_owners`. If `bucket_owners` is configured and a bucket is not included in one of the mapped configurations, `default_bucket_owner` defaults to the account ID in `aws.sts_role_arn`. You can configure both `bucket_owners` and `default_bucket_owner` and apply the settings together.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:203]:
```suggestion
The `parquet` codec writes data using the schema. Because Parquet requires an Avro schema, you may either define the schema yourself or have Data Prepper automatically generate it. Defining your own schema is recommended, as this will allow it to be tailored to your particular use case.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (natebower) [inline _data-prepper/pipelines/configuration/sinks/s3.md:174]: Is "in order for the automatically generated schema to produce a working schema" redundant? Do we just mean "in order to automatically generate a working schema"?

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (natebower) [inline _data-prepper/pipelines/configuration/sinks/s3.md:176]:
```suggestion
Avro fields should use a null [union](https://avro.apache.org/docs/current/specification/#unions) because this will allow missing values. Otherwise, all required fields must be present for each event. Use non-nullable fields only when you are certain they exist.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (natebower) [inline _data-prepper/pipelines/configuration/sinks/s3.md:170]:
```suggestion
The `avro` codec writes an event as an [Apache Avro](https://avro.apache.org/) document. Because Avro requires a schema, you may either define the schema or have Data Prepper automatically generate it. Defining your own schema is recommended because this will allow it to be tailored to your particular use case.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (natebower) [inline _data-prepper/pipelines/configuration/sinks/s3.md:203]:
```suggestion
The `parquet` codec writes data using the schema. Because Parquet requires an Avro schema, you may either define the schema yourself or have Data Prepper automatically generate it. Defining your own schema is recommended because this will allow it to be tailored to your particular use case.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (natebower) [inline _data-prepper/pipelines/configuration/sinks/s3.md:205]:
```suggestion
For more information about the Avro schema, refer to [Avro codec](#avro-codec).
```

**Review by natebower — COMMENTED** [review]
@vagimeli @graytaylor0 Please see my comments and changes and let me know if you have any questions. Thanks!

**Review by dlvenable — APPROVED** [review]

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:123]:
```suggestion
`sts_role_arn` | No | String | The AWS Security Token Service (AWS STS) role to assume for requests to Amazon Simple Queue Service (Amazon SQS) and Amazon S3. Defaults to `null`, which uses [standard SDK behavior for the credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html).
```

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [reply _data-prepper/pipelines/configuration/sinks/s3.md]: Yes. Revised.

File: _data-prepper/pipelines/configuration/sinks/s3.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/s3.md:174]:
```suggestion
In cases where your data is uniform, you may be able to automatically generate a schema. Automatically generated schemas are based on the first event that the codec receives. The schema will only contain keys from this event, and all keys must be present in all events in order to automatically generate a working schema. Automatically generated schemas make all fields nullable. Use the `include_keys` and `exclude_keys` sink configurations to control which data is included in the automatically generated schema.
```

**Review by vagimeli — APPROVED** [review]
Doc review complete. Editorial review feedback and edits accepted. Revised accordingly.

**Pull request merged**
